src/server/game/Entities/Creature/CreatureData.h

```diff
@@ -564,40 +564,45 @@ struct CreatureAddon
 // Vendors
 struct VendorItem
 {
-    VendorItem(uint32 _item, int32 _maxcount, uint32 _incrtime, uint32 _ExtendedCost, uint8 _Type)
-        : item(_item), maxcount(_maxcount), incrtime(_incrtime), ExtendedCost(_ExtendedCost), Type(_Type) { }
+    VendorItem() : item(0), maxcount(0), incrtime(0), ExtendedCost(0), Type(0), PlayerConditionId(0), IgnoreFiltering(false) {};
 
     uint32 item;
     uint32 maxcount;                                        // 0 for infinity item amount
     uint32 incrtime;                                        // time for restore items amount if maxcount != 0
     uint32 ExtendedCost;
     uint8  Type;
+    std::vector<int32> BonusListIDs;
+    uint32 PlayerConditionId;
+    bool IgnoreFiltering;
 
     //helpers
     bool IsGoldRequired(ItemTemplate const* pProto) const;
 };
 
 struct VendorItemData
 {
-    std::vector<VendorItem> m_items;
+    std::vector<VendorItem const*> m_items;
```

Comment:
Why did you make this a pointer again?

Reply:
With the way the rest of the code is organized, it just seemed more efficient instead of copying the whole VendorItem object into the vector. But I could be wrong on that, was there a better reason it wasn't a pointer?

Reply:
Iterating it is more efficient - and that is the majority of uses, adding/removing isn't something done very often.

Reply:
Fair point, I'll revert the pointer change